app/models/attributable/attribute.rb

```diff
@@ -180,9 +186,15 @@ def to_field_info(validator_source = nil)
         kind: kind,
         required: required?
       }
+
       options.update(selection: selection_options(validator_source)) if selection?
-      options.update(step: 1, min: minimum) if integer?
-      options.update(step: 0.1, min: 0) if float?
+      options_hash = { min: minimum }
+
+      options_hash.update(max: maximum) if maximum.present?
+      options_hash.update(step: 1) if integer?
+      options_hash.update(step: 0.1) if float?
+
+      options.update(**options_hash) if options_hash.present?
```

Comment:
options_hash is always present because it is initialised with `options_hash = { min: minimum }` above.